cmake/CPM.cmake

```diff
@@ -790,6 +798,12 @@ function(CPMAddPackage)
     cpm_get_fetch_properties("${CPM_ARGS_NAME}")
   endif()
 
+  if(CPM_HAS_CACHE_LOCK)
+    file(LOCK ${download_directory}/../cmake.lock RELEASE)
+    file(REMOVE ${download_directory}/../cmake.lock)
+    set(CPM_HAS_CACHE_LOCK OFF)
+  endif()
```

Comment:
Would this break when a dependency itself uses CPM to download additional dependencies? 

Afaict inside the inner `CPMAddPackage` call, `CPM_HAS_CACHE_LOCK` would be inherited from the outer scope, resulting in them not creating a lock, but releasing and removing any locks relative to their download directory.

Maybe it could be placed before the `cpm_add_subdirectory` call?

Reply:
There are two `cpm_add_subdirectory` calls in the area between the file locking and unlocking. Which one are you thinking it should be before? If it goes before both of them, then there are calls to `cpm_fetch_package` that would be unprotected.

Reply:
I took a stab at solving this problem by changing `CPM_HAS_CACHE_LOCK` to `${CPM_ARGS_NAME}_CPM_HAS_CACHE_LOCK`. This should create a new variable for each new package, preventing the variable that controls protection from being the same one that is used in the next scope.

Reply:
Understood, I believe that should work as well. Thanks for the changes!